wordpress-mobile/WordPress-Android
# Fix upload notification not shown on retry

Fixes #9951 

This PR takes care of promoting the UploadService to the foreground (shows an upload in progress notification) when retrying upload.

I've removed this line
`mPostUploadNotifier.removePostInfoFromForegroundNotificationData(post, mediaToRetry);`
but tbh I'm not sure what was the initial purpose of the line. I haven't find any reason why we need to invoke it on this place. If I keep it there, the upload notification will always show "1 post remaining" even when I invoke retry on three posts.
Imagine we have 3 posts
1. First post increases `sNotificationData.mTotalPostItems` to 1
2. Second post invokes `mPostUploadNotifier.removePostInfoFromForegroundNotificationData(post, mediaToRetry);` -> decreases `sNotificationData.mTotalPostItems` to 0 -> and increases it back to 1
3. Third post also decreases it to 0 and back to 1
4. The notification shows "1 post remaining" even when there are actually 3 posts being uploaded


To test:
![retry-notification-2](https://user-images.githubusercontent.com/2261188/58540843-78ccb700-81fa-11e9-97a3-9f99e03a8c20.gif)
1. Turn on airplane mode
2. Create a local draft
3. Turn off airplane mode and notice the upload notification is being shown
--------------
1. Turn on airplane mode
2. Create 3 local drafts
3. Turn off airplane mode and notice the upload notification (3 posts remaining) is being shown
--------------
![retry-notification](https://user-images.githubusercontent.com/2261188/58541014-d4974000-81fa-11e9-9643-4b648791dc69.gif)

1. Open editor
2. Enter title
3. Press back and turn on airplane mode asap
4. Wait until the upload fails
5. Turn off airplane mode and notice the upload notification is shown
----------------
1. Open editor and enter title
2. Press back and turn on airplane mode asap
3. Open editor and enter title
4. Turn off airplane mode, press back and turn on airplane mode (you might want to set network type to GPRS in emulator settings so the draft from step 2 doesn't get uploaded)
5. Wait until the upload fails
6. Turn off airplane mode and notice the upload notification (2 posts remaining) is shown

Update release notes:
